src/code/z_snap.c

```diff
@@ -119,42 +146,48 @@ s32 func_8013A530(PlayState* play, Actor* actor, s32 flag, Vec3f* pos, Vec3s* ro
     s32 ret = 0;
     s32 bgId;
 
+    // Check distance
     distance = OLib_Vec3fDist(pos, &camera->eye);
     if ((distance < distanceMin) || (distanceMax < distance)) {
-        func_8013A41C(0x3F);
-        ret = 0x3F;
+        Snap_SetFlag(PICTOGRAPH_BAD_DISTANCE);
+        ret = PICTOGRAPH_BAD_DISTANCE;
     }
 
-    x = func_8013A504(Camera_GetCamDirPitch(camera) + rot->x);
-    y = func_8013A504(Camera_GetCamDirYaw(camera) - BINANG_SUB(rot->y, 0x7FFF));
-    if ((0 < angleError) && ((angleError < x) || (angleError < y))) {
-        func_8013A41C(0x3E);
-        ret |= 0x3E;
+    // Check rot is facing camera?
+    x = Snap_AbsS(Camera_GetCamDirPitch(camera) + rot->x);
+    y = Snap_AbsS(Camera_GetCamDirYaw(camera) - BINANG_SUB(rot->y, 0x7FFF));
+    if ((0 < angleRange) && ((angleRange < x) || (angleRange < y))) {
+        Snap_SetFlag(PICTOGRAPH_BAD_ANGLE);
+        ret |= PICTOGRAPH_BAD_ANGLE;
     }
 
-    Actor_GetProjectedPos(play, pos, &screenSpace, &distance);
-    x = (s16)(screenSpace.x * distance * 160.0f + 160.0f) - 85;
-    y = (s16)(screenSpace.y * distance * -120.0f + 120.0f) - 67;
-    if ((x < 0) || (150 < x) || (y < 0) || (105 < y)) {
-        func_8013A41C(0x3D);
-        ret |= 0x3D;
+    // Check in viewfinder
+    Actor_GetProjectedPos(play, pos, &projectedPos, &distance);
+    x = (s16)(projectedPos.x * distance * 160.0f + 160.0f) - 85;
+    y = (s16)(projectedPos.y * distance * -120.0f + 120.0f) - 67;
+    if ((x < 0) || (x > 150) || (y < 0) || (y > 105)) {
```

Comment:
```suggestion
    x = (s16)(projectedPos.x * distance * (SCREEN_WIDTH / 2) + (SCREEN_WIDTH / 2)) - ((SCREEN_WIDTH / 2) + 5);
    y = (s16)(projectedPos.y * distance * -(SCREEN_HEIGHT / 2) + (SCREEN_HEIGHT / 2)) - ((SCREEN_HEIGHT / 2) + 7);
    if ((x < 0) || (x > (SCREEN_WIDTH - 10)) || (y < 0) || (y > (SCREEN_HEIGHT - 15))) {
```
Might as well go the full way t onot worry about conflicts. Some of these literals seem weird though, specifically the shift of `85` and `67`. Do you think my interpretation of those constants are right?

Note: without the shifts, you directly get screen coordinates
```
x = (s16)(projectedPos.x * distance * (SCREEN_WIDTH / 2) + (SCREEN_WIDTH / 2));
y = (s16)(projectedPos.y * distance * -(SCREEN_HEIGHT / 2) + (SCREEN_HEIGHT / 2));
```

Reply:
As discussed in Discord, I'll make this a TODO rather than attempting to nail it down now.